HackAtUCI/irvinehacks-site-2024
# Integrate waitlist promotion modal and action

Finishes the remaining of #340.

## Changes
- Create new promote action that shows modal with confirmation button and instructions to participant
- Show promote action instead of check in for waitlisted participants
- Disable promote action for organizers who are not check-in leads, showing [PopOver](https://cloudscape.design/components/popover/) message
- Disable check in action for participants with `WAIVER_SIGNED` status, showing PopOver message

## Testing
1. Impersonate yourself and navigate to the participants page of the admin dashboard. Make sure to grant yourself the director role!
2. If they don't already exist, create records of users with the `ATTENDING`, `WAIVER_SIGNED`, `WAITLISTED`, and `ACCEPTED` statuses.
3. Click "Check In" and ensure the modal has popped up. Then click "Confirmed". A new check-in record should show up in the database.
4. Ensure that you **do not** have access to click "Promote" and that when clicking on it, a tooltip shows up explaining why.
5. Now grant your UID the role of `checkin_lead` to allow yourself to check in people and promote applicants off the waitlist.
6. Click "Promote". The API call should return a 500 status code because this route updates the applicant's status to `ACCEPTED` and then attempts to send out an email to the applicant notifying them of their acceptance, but you do not have the API key for that.
